### PR TITLE
Adding plotting to Muon Analysis 2

### LINKED ISSUE #25738: Muon Analysis 2 multi plotting
Want to add plotting to the new muon GUI

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This PR adds plotting to muon analysis 2. This is currently simple plotting of a single workspace. It also updates the multiplotting widget to work in workbench.

**Report to:** Peter

**To test:**
  * Ideally test in both workbench and mantidplot
  * Open Muon Analysis 2, when you load some data a plot window should pop up.
  * If you then do a fit on this data the fit should be added to the plot. 
  * Loading multiple runs should add all the runs to the graph and a fit on any of these graphs should then be added automatically. 
  * Check that plotting works correctly in frequency domain analysis
  * In mantidplot check the elemental analysis plotting still functions correctly.

Fixes #25738  

*This does not require release notes* because this adds functionality to a new interface which is having it's release notes done separately.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/MultiPlotting/multi_plotting_widget.py [hunk 1]
```diff
@@ -11,31 +11,41 @@
...
+        self.figure = Figure()
+        self.figure.set_facecolor("none")
+        self.canvas = FigureCanvas(self.figure)
+        self.close_callback = close_callback
+
         self.plot_context = PlottingContext()
-        self.multi_plot = MultiPlotWidget(self.plot_context, self)
-        self.setCentralWidget(self.multi_plot)
-        self.setWindowTitle(window_title)
+        self.multi_plot = MultiPlotWidget(window_title, self.plot_context, self.canvas)
```
Comment (AnthonyLim23) [inline scripts/MultiPlotting/multi_plotting_widget.py]: should we just replace `self.multi_plot` with `self.window`?

Comment (Mantid-Matthew) [reply scripts/MultiPlotting/multi_plotting_widget.py]: In the medium term we should, I left it with both to minimize the chances of breaking existing code which relies on the old name but having thought a bit more that probably only happens in a couple of places so can probably just update now.

**Comment by AnthonyLim23** [comment]
In elemental analysis. Load some data and plot it. Then try and delete one of the workspaces and you get an error

```
Traceback (most recent call last):
  File "c:\users\btr75544\work\dev\mantid\qt\python\mantidqt\plotting\ads_observer.py", line 30, in wrapper
    func(*args, **kwargs)
  File "c:\users\btr75544\work\dev\mantid\qt\python\mantidqt\plotting\ads_observer.py", line 80, in deleteHandle
    ax.remove_workspace_artists(workspace)
  File "c:\users\btr75544\work\dev\mantid\framework\pythoninterface\mantid\plots\__init__.py", line 317, in remove_workspace_artists
    workspace_artist.remove(self)
  File "c:\users\btr75544\work\dev\mantid\framework\pythoninterface\mantid\plots\__init__.py", line 99, in remove
    self._remove(axes, self._artists)
  File "c:\users\btr75544\work\dev\mantid\framework\pythoninterface\mantid\plots\__init__.py", line 124, in _remove
    artist.remove()
  File "C:\Users\BTR75544\work\dev\mantid\external\src\ThirdParty\lib\python2.7\lib\site-packages\matplotlib\artist.py", line 151, in remove
    self._remove_method(self)
  File "C:\Users\BTR75544\work\dev\mantid\external\src\ThirdParty\lib\python2.7\lib\site-packages\matplotlib\axes\_base.py", line 1964, in <lambda>
    line._remove_method = lambda h: self.lines.remove(h)
ValueError: list.remove(x): x not in list
```

**Comment by Mantid-Matthew** [comment]
Closing this in favor of #25903, would like to keep the branch around for now though.

**Pull request closed without merging**